tools/generate_events.py

```diff
@@ -0,0 +1,268 @@
+type {{eventName}} struct {
+{%- for p, t in event.properties.items() %}
+    {{event.pascal_case(p)}} {{event.go_type(t)}}
+{%- endfor %}
+}
+
+var {{eventName}}Name string = "{{event['name']}}"
+
+func New{{eventName}}(
+{%- for p, t in event.properties.items() %}
+    {{event.camel_case(p)}} {{event.go_type(t)}},
+{%- endfor %}
+) *{{eventName}} {
+    return &{{eventName}}{
+{%- for p, t in event.properties.items() %}
+        {{event.pascal_case(p)}}: {{event.camel_case(p)}},
+{%- endfor %}
+    }
+}
+
+func (e *{{eventName}}) GetName() string {
+    return "{{event['name']}}"
+}
+
+func (e *{{eventName}}) GetSeverity() string {
+    {% if event.properties.severity -%}     return e.Severity
+    {%- else -%}      return "{{event.event_severity()}}"
+    {%- endif %}
+}
+
+func (e *{{eventName}}) GetClusterId() *strfmt.UUID {
+    return &e.ClusterId
+}
+
+{%- if event.type != "cluster" %}
+func (e *{{eventName}}) GetHostId() *strfmt.UUID {
+    return &e.HostId
+}
+{%- endif %}
+
+func (e *{{eventName}}) format(message *string) string {
+    r := strings.NewReplacer(
+{%- for p, t in event['properties'].items() %}
+        "{{'{'}}{{p}}{{'}'}}", fmt.Sprint(e.{{event.pascal_case(p)}}),
+{%- endfor %}
+    )
+    return r.Replace(*message)
+}
+
+func (e *{{eventName}}) FormatMessage() string {
+    s := "{{event.format}}"
+    return e.format(&s)
+}
+
+{% endfor -%}
+''')
+
+
+class EventGenerator(object):
+    VALID_PROPS_TYPES = {"UUID":dict(go_type="strfmt.UUID", go_import="github.com/go-openapi/strfmt"),
+                         "integer":dict(go_type="int", go_import=None),
+                         "int64":dict(go_type="int64", go_import=None),
+                         "string":dict(go_type="string", go_import=None),
+                         "bool":dict(go_type="bool", go_import=None)}
+    def __init__(self, event):
+        self.event = event
+
+    def __getattr__(self, attr):
+        #import ipdb; ipdb.set_trace()
+        if attr in self.__dict__:
+            return getattr(self, attr)
+        else:
+            return getattr(self.event, attr)
+
+    def ctor_name(self):
+        return "New" + self.event_class()
+
+    def ctor_args(self):
+        args = []
+        for arg, _ in self.event.properties.iteritems():
+            args.append(EventGenerator.camel_case(arg) + " " + EventGenerator.pascal_case(arg))
+        return ", ".join(args)
+
+    def event_class(self):
+        return EventGenerator.pascal_case(self.event.name) + "Event"
+
+    def event_type(self):
+        return "Type" + EventGenerator.pascal_case(self.event.name)
+
+    def event_severity(self):
+        if self.event.severity.lower() == "info":
+            return "info"
+        if self.event.severity.lower() == "warning":
+            return "warning"
+        if self.event.severity.lower() == "error":
+            return "error"
+        if self.event.severity.lower() == "critical":
+            return "critical"
```

Comment:
Why not just
```suggestion
    return self.event.severity.lower() 
```

Reply:
Fixed and added validation of supported severity values in 'validate_event' function.